analysis/wwz/wwz4l.py

```diff
@@ -452,9 +477,31 @@ def process(self, events):
 
             # Clean with dr (though another option is to use jetIdx)
             cleanedJets = os_ec.get_cleaned_collection(l_wwz_t,jets)
+            jetptname = "pt_nom" if hasattr(cleanedJets, "pt_nom") else "pt"
+
+            ##### JME Stuff #####
+
+            cleanedJets["pt_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.pt
+            cleanedJets["mass_raw"] = (1 - cleanedJets.rawFactor)*cleanedJets.mass
+
+            if not isData:
+                cleanedJets["pt_gen"] =ak.values_astype(ak.fill_none(cleanedJets.matched_gen.pt, 0), np.float32)
+            else:
+                cleanedJets["pt_gen"] =ak.ones_like(cleanedJets.pt)
+
+            if year in ["2022","2022EE","2023","2023BPix"]:
+                cleanedJets["rho"] = ak.broadcast_arrays(events.Rho.fixedGridRhoFastjetAll, cleanedJets.pt)[0]
```

Comment:
What does this broadcast_arrays do, can you add a comment to explain? Also, can you just use the `rho` that you defined at the top?

Reply:
added comment, but will explain here as well:

this broadcasts the rho variable to have the same shape as correctedJets

Reply:
Thanks for the comment. 

But for the second part of the comment, can we not just use the `rho` you defined at the top of the processor? Why do you need to do events.Rho.etc here too?

Reply:
Oh you did replace with `rho`. Ok, so now that the if and elif do exactly the same thing here, we can get rid of if/elif and just do the thing